src/V3Randomize.cpp

```diff
@@ -725,11 +732,35 @@ class ConstraintExprVisitor final : public VNVisitor {
                     memberp = VN_CAST(memberp->nextp(), MemberDType);
             }
         }
+        // Mark Random for structArray
+        if (VN_IS(nodep->fromp(), ArraySel)) {
+            AstNodeExpr* const fromp = VN_AS(nodep->fromp(), ArraySel)->fromp();
+            VN_AS(fromp->dtypep()->skipRefp()->subDTypep()->skipRefp(), StructDType)
+                ->markConstrainedRand(true);
+            AstMemberDType* memberp
+                = VN_AS(fromp->dtypep()->skipRefp()->subDTypep()->skipRefp(), StructDType)
```

Comment:
Please make a temp var with the VN_AS result, since skipRefp() is slow so no need to call twice.